src/persian_content/mod.rs

```diff
@@ -7,15 +7,42 @@ lazy_static! {
 }
 
 /// Set of helpers for manipulating Persian text.
-trait PersianContent: AsRef<str> {
+pub trait PersianContent: AsRef<str> {
     /// Checks if a text has at least a Persian char in it.
     fn has_persian_char(&self) -> bool {
         HAS_PERSIAN_CHAR.is_match(self.as_ref())
     }
 
     /// Checks if a text is in Persian.
     fn is_persian_str(&self) -> bool {
-        PERSIAN_STR.is_match(self.as_ref())
+        // First remove the non-alphabetic chars
+        let str = self
```

Comment:
This code has both congestion and performance issues I think we'd better fix Regex

Reply:
Any solution?

Reply:
`^[\u0600-\u06FF\s.,?!-_]+$`

Reply:
So what about emojis? What about allllllll other unicode chars that are not alphabetic?

Reply:
`^[\u0600-\u06FF]|[[:punct:]]+$`

Reply:
about emoji I have no idea

Reply:
There are about 12350 non-alphabetic chars in unicode, ref: https://www.compart.com/en/unicode/category

Reply:
And you know, u0600-u06FF is not correct! :D

Reply:
Okay, so we'll merge this code, thanks